weatherv3/www/js/controllers.js

```diff
@@ -0,0 +1,116 @@
+      $scope.precipType = current.currently.precipType;
+      $scope.nearestStormDistance = current.currently.nearestStormDistance;
+
+      console.log('tempDif: ', $scope.tempDif);
+      console.log('GOT CURRENT', current);
+      //debugger;
+    }, function(error) {
+      alert('Unable to get current conditions');
+      console.error(error);
+    });
+
+})
+
+.controller('weekCtrl', function($scope,$state,$filter,Weather,DataStore,WeekDays) {
+    //read default settings into scope
+    console.log('inside weekCtrl');
+    $scope.city  = DataStore.city;
+    var latitude  =  DataStore.latitude;
+    var longitude = DataStore.longitude;
+
+    //call getCurrentWeather method in factory ‘Weather’
+    Weather.getCurrentWeather(latitude,longitude).then(function(resp) {
+      $scope.current = resp.data;
+      console.log('GOT CURRENT', $scope.current);
+      theForecast();
+      //debugger;
+    }, function(error) {
+      alert('Unable to get current conditions');
+      console.error(error);
+    });
+
+    var weekDays = WeekDays;
+    var today = WeekDays.getToday();
+    //returns an array with info for the next 7 days
+    var theForecast = function() {
+      console.log('in theForecast function. value of current: ', $scope.current);
+      var d = new Date();
+      result = new Array(7);
+      var n = today;
+
+      $scope.barTypes = ['bar-calm', 'bar-balanced', 'bar-energized', 'bar-assertive', 'bar-royal', 'bar-dark','bar-positive', 'bar-calm', 'bar-balanced', 'bar-energized', 'bar-assertive', 'bar-royal', 'bar-dark','bar-positive'];
+      $scope.barFirst = n;
+      for (i = 0; i < 7; i++) {
+          var ctempMax = "High: " + Math.round($scope.current.daily.data[i].temperatureMax) + "°F, ";
+          var ctempMin = "Low: " + Math.round($scope.current.daily.data[i].temperatureMin) + "°F, ";
+          var cweatherSummary = $scope.current.daily.data[i].summary.toLowerCase();
+          
+          var cday = WeekDays[n] + ",";
+          var cdate = $filter('date')(d);
+          result[i] = {date: cdate, tempMax: ctempMax, tempMin: ctempMin, weatherSummary:cweatherSummary, day:cday};
+          
+          if (n < 6) {
+            n++;
+          }
+          else {
+            n = 0;
+          }
+          d.setDate(d.getDate() + 1);
+
+      };
+      $scope.forecast = result;
+    };
+})
+
+.controller('LocationsCtrl', function($scope,$state, Cities,DataStore) {
+  $scope.cities = Cities.all();
+
+  $scope.changeCity = function(cityId) {
+    //get lat and longitude for seleted location
+    var lat  = $scope.cities[cityId].lat; //latitude
+    var lgn  = $scope.cities[cityId].lgn; //longitude
+    var city = $scope.cities[cityId].name; //city name
+
+    DataStore.setCity(city);
+    DataStore.setLatitude(lat);
+    DataStore.setLongitude(lgn);
+    console.log("hopefully: city: " + city + ", lat: " + lat);
+    console.log("$scope.city: " + DataStore.city);
+    $state.go('tab.home');
+  }
+})
```

Comment:
Cool, just 1 suggestion, maybe putting your controllers in a separate file to optimize organization could be helpful